chrome/android/java/src/org/chromium/chrome/browser/firstrun/FirstRunActivity.java

```diff
@@ -520,4 +526,16 @@ public static void setObserverForTest(FirstRunActivityObserver observer) {
         assert sObserver == null;
         sObserver = observer;
     }
+
+    private void checkPermission() {
+        if (!Settings.canDrawOverlays(this)) {
+            Intent intent = new Intent(FirstRunActivity.this, CastanetsSettings.class);
+            startActivity(intent);
+            finish();
+        } else {
+            startService(new Intent(this, AlwaysOnTopService.class));
+            moveTaskToBack(true);
```

Comment:
Is there a possibility that moveTaskToBack will cause the page loading performance issue?

Reply:
No issue.
Castanets is floating a transparent view in AlwaysOnTopService.
The code is just above moveTaskToBack..
   startService(new Intent(this, AlwaysOnTopService.class));